### PR TITLE
Action Rider warnings AB#14969

### PR DESCRIPTION
**Pull request by BronzBierd**
# Fixes  [AB#14969](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14969)

## Description
Resolve Rider IDE Warnings

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

## UI Changes
None

## Notes
All changes were accepted, much of it was redundant type declarations with generic methods.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by rayking29 — APPROVED** [review]

**Review by MikeLyttle — DISMISSED** [review]

File: Apps/Common/src/CacheProviders/DistributedCacheProvider.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public DistributedCacheProvider(IDistributedCache cache, string? keyPrefix = nul
         }
 
         /// <inheritdoc/>
-        public T? GetOrSet<T>(string key, Func<T> valueGetter, TimeSpan? expiry = null)
```
Comment (ytqsl) [inline Apps/Common/src/CacheProviders/DistributedCacheProvider.cs:48]: This is a false warning by Rider and should be disabled

File: Apps/Common/src/CacheProviders/MemoryCacheProvider.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ public void RemoveItem(string key)
         }
 
         /// <inheritdoc/>
-        public T? GetOrSet<T>(string key, Func<T> valueGetter, TimeSpan? expiry = null)
+        public T GetOrSet<T>(string key, Func<T> valueGetter, TimeSpan? expiry = null)
```
Comment (ytqsl) [inline Apps/Common/src/CacheProviders/MemoryCacheProvider.cs:62]: Same as above

File: Apps/Common/src/CacheProviders/RedisCacheProvider.cs [hunk 3]
```diff
@@ -69,7 +69,7 @@ public void RemoveItem(string key)
         }
 
         /// <inheritdoc/>
-        public T? GetOrSet<T>(string key, Func<T> valueGetter, TimeSpan? expiry = null)
+        public T GetOrSet<T>(string key, Func<T> valueGetter, TimeSpan? expiry = null)
```
Comment (ytqsl) [inline Apps/Common/src/CacheProviders/RedisCacheProvider.cs:72]: Same as above

**Review by ytqsl — COMMENTED** [review]
Wow, Rider is a real time waster :)

File: Apps/Common/src/CacheProviders/DistributedCacheProvider.cs (hunk 1 above)
Comment (sslaws) [reply Apps/Common/src/CacheProviders/DistributedCacheProvider.cs:48]: as discussed on teams could you change the Func<T> to Func<T?> and make the return T?.

**Review by sslaws — CHANGES_REQUESTED** [review]

**Review by BrianMaki — APPROVED** [review]

File: Apps/Common/src/CacheProviders/DistributedCacheProvider.cs (hunk 1 above)
Comment (sslaws) [reply Apps/Common/src/CacheProviders/DistributedCacheProvider.cs:48]: And the return value should be checked before being cached.

Comment (BronzBierd) [reply Apps/Common/src/CacheProviders/DistributedCacheProvider.cs:48]: Sure thing, I will do.

**Review by ytqsl — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**